.github/workflows/main.yaml

```diff
@@ -41,14 +41,23 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Create branch from commit by event name
+        run: |
+          if [[ '${{ github.event_name }}' == 'push' ]]; then
+            git branch __branch_before ${{ github.event.before }}
+          elif [[ "${{ github.event_name }}" == 'pull_request' ]]; then
+            git branch __branch_before ${{ github.event.pull_request.base.sha }}
+          fi
       - name: Set up JDK ${{ matrix.java_version }}
         uses: actions/setup-java@v4
         with:
           java-version: ${{ matrix.java_version }}
           distribution: 'temurin'
           cache: 'maven'
       - name: Build with JDK ${{ matrix.java_version }}
-        run: mvn --batch-mode install -DskipITs ${{ matrix.included_modules }}
+        run: mvn --batch-mode install ${{ matrix.included_modules }} -Dgib.disable=false -Dgib.referenceBranch=__branch_before
```

Comment:
Those `-pl !...` options in `included_modules` probably mandate [`disableSelectedProjectsHandling`](https://github.com/gitflow-incremental-builder/gitflow-incremental-builder?tab=readme-ov-file#gibdisableselectedprojectshandling).

I'll have to take a closer look.